Eating deep fried inedible items moves them to stomach

cacogen opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

- Having inedibles in stomach increases disgust at random intervals based on the amount
- Vomiting empties stomach contents
- Can squeeze out contents of stomachs in hand
- Stomachs only hold a combined weight class of 15 (so 5 normal sized objects)
- Updates toy singulo suicide to use stomach instead of cavity implants

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Stops people using the publicly available deep fryers to brick or delete items by instead moving the item to the eater's stomach which is a semi-difficult place to retrieve them from and leads to all sorts of wacky hijinks trying to do it. Randomly vomiting up a slew of items is funny too.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
balance: Deep fried inedible items move to stomach when eaten instead of being deleted 
add: Having inedibles in stomach increases disgust and/or damages stomach at random intervals based on the amount
add: Vomiting empties stomach contents
add: Can squeeze out contents of stomachs in hand
expansion: Toy singulo suicide uses stomach instead of cavity implants
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

cacogen committed:
Eating deep fried inedibles moves them to stomach

- Having inedibles in stomach every so often increases disgust based on the amount
- Vomiting empties stomach contents
- Can squeeze out contents of stomachs in hand

Zonespace27 reviewed (commented):
Length over .len

cacogen committed:
length()

Co-authored-by: Zonespace <41448081+Zonespace27@users.noreply.github.com>

MrDoomBringer commented on code/modules/surgery/organs/stomach/_stomach.dm:
```diff
@@ -205,6 +217,18 @@
 	return !(NOSTOMACH in owner_species.inherent_traits)
 
+/obj/item/organ/stomach/proc/handle_contents(mob/living/carbon/human/mule, delta_time)
+	var/inedibles = 0
```
```suggestion
	var/inedibles = 0
	if(is_clown_job(mule.mind?.assigned_role)))
		return
```
through years of classical stage training, clowns have learned how to handle eating inedible things

cacogen replied:
You can't deep fry the disk. Although you could if eating it wouldn't destroy it.

MrDoomBringer reviewed (changes requested):
Quick deep-fry the disk and feed it to the clown

Iamgoofball commented:
:-1: 
massive deep fryer nerf

cacogen commented:
I'm not sure how this got made about vore fetishism but there seems to be a fundamental misunderstanding of what that actually is here.

moocowswag commented:
>Having inedibles in stomach increases disgust at random intervals based on the amount

Does it take the size of the item into account or just how many items there are or both.

MrDoomBringer commented:
> Does it take the size of the item into account 

no

IndieanaJones commented:
The security came inside and found the captain, dead with a HUGE belly full of MMIs.

cacogen commented:
Yeah, that's pretty funny though. You shouldn't be able to eat mobs in the first place (at least not ones with minds) but I wouldn't take the opportunity to do that away from people.

cacogen committed:
Adds some boring and unnecessary checks for volume

ATH1909 commented:
sad

Farquaar commented:
[Squeaky Boots Energy](https://youtu.be/7bMhiSQMY1M)

itseasytosee commented:
> sad
> 
> now people will be lynched for trying to serve deep fried grenades or w/e

They will never expect the deep-fried fedora with a grenade inside.

Bawhoppen commented:
👍 x 💯

tralezab commented on code/game/objects/items/food/bread.dm:
```diff
@@ -325,6 +325,18 @@
 
 /obj/item/food/deepfryholder/proc/On_Consume(eater, feeder)
 	if(contents)
+		if(isliving(eater))
```
early return both of these (second check should `QDEL_LIST(contents)` before returning)

tralezab commented on code/game/objects/items/food/bread.dm:
```diff
@@ -325,6 +325,18 @@
+		if(isliving(eater))
+			var/mob/living/guy_fieri = eater
+			var/obj/item/organ/stomach/stomach = guy_fieri.getorganslot(ORGAN_SLOT_STOMACH)
+			if(stomach)
```
early return this as well

tralezab commented on code/modules/surgery/organs/stomach/_stomach.dm:
```diff
@@ -39,6 +40,16 @@
+/obj/item/organ/stomach/attack_self(mob/user, modifiers)
+	. = ..()
+
+	if(length(contents))
```
early return

tralezab commented on code/modules/surgery/organs/stomach/_stomach.dm:
```diff
@@ -39,6 +40,16 @@
+
+	if(length(contents))
+		to_chat(user, span_notice("You begin squeezing out the contents of [src]..."))
+		if(do_after(user, length(contents) SECONDS / 2))
```
early return

tralezab commented on code/modules/surgery/organs/stomach/_stomach.dm:
```diff
@@ -205,6 +218,25 @@
+/obj/item/organ/stomach/proc/handle_contents(mob/living/carbon/human/mule, delta_time)
+	var/inedibles = 0
+	for(var/atom/movable/thing in contents)
+		if(!IsEdible(thing))
```
early continue. if it's edible, move onto the next thing.

tralezab reviewed (changes requested):
your early return game my friend, it must be practiced🙏

tralezab commented:
otherwise cool brohan

ATH1909 commented:
how will this interact with the toy singulo suicide?

optimumtact commented:
What happens if they have no stomach organ or their species takes no disgust?

you need a storage cap and an auto vomit at some level of items.

cacogen committed:
Makes spinning toy suicide work with this

- Adds stomach damage at random intervals based on volume held in stomach
- Adds autovomit when stomach is too full

cacogen committed:
Sins of a solar empire

cacogen committed:
how did you get your early return game to come up

cacogen commented:
I don't think you can eat without a stomach but if you manage it, it just deletes the contents of the deepfry holder like it used to. I've added auto vomit after a certain (low) combined volume of objects too now.

>how will this interact with the toy singulo suicide?

I've updated the suicide to avoid its regurgitation being doubled.

cacogen committed:
Improves timing of singulo toy suicide

- Adds feedback for emptying empty stomach
- Adds examine message explaining how to empty stomachs

cacogen committed:
Different volumes for different stomachs

This feels like making my own shitty imitation storage object on the side of real storage

cacogen commented:
![](https://i.imgur.com/l1KnkLv.png)

@tralezab 
plz help
